py/desitarget/secondary.py

```diff
@@ -636,6 +641,15 @@ def finalize_secondary(scxtargs, scnd_mask, sep=1., darkbright=False):
     scxtargs = rfn.rename_fields(
         scxtargs, {'SCND_TARGET': prepend+'SCND_TARGET'}
     )
+
+    # APC Remove duplicate targetids from secondary-only targets
+    alldups = []
+    for _, dups in duplicates(scxtargs['TARGETID']):
+        dups = np.delete(dups, 0) # Retain the first
```

Comment:
I don't think we want to retain the first duplicate among matching secondary targets, I think we want to retain the duplicate with the _highest priority_. Is it possible to rewrite this to retain the highest priority secondary-only target?

Reply:
Good catch, thanks -- will sort this out. Thanks for the review!

Reply:
Done, except I wasn't quite sure what was meant in the docstring by "the first encountered" . I wrote it to use the lowest index in the original table to break ties.

Reply:
Oh, "the first encountered" here just means that if we find two secondaries with the same priority, we just keep the first one in the table/array/loop. In other words, we don't break ties in any clever way once we've broken ties to keep the highest-priority secondary. That's not critical though. The way we write everything back to the input files and generate consistent TARGETIDs should keep track of what was matched and how.

If you _don't_ think the code does this, just remove the sentence in the docstring. As I noted, it's not critical how we break ties other than on priority.

Reply:
OK, in that case I'll just use np.argmax directly, since that breaks ties on the lowest index in the list of duplicates. Done now.